connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/protobuf/functions.scala

```diff
@@ -199,4 +301,16 @@ object functions {
       options: java.util.Map[String, String]): Column = {
     fnWithOptions("to_protobuf", options.asScala.iterator, data, lit(messageClassName))
   }
+
+  private def emptyOptions: java.util.Map[String, String] = Collections.emptyMap[String, String]()
+
+  private def readDescriptorFileContent(filePath: String): Array[Byte] = {
+    // This method is copied from org.apache.spark.sql.protobuf.util.ProtobufUtils
+    try {
+      FileUtils.readFileToByteArray(new File(filePath))
+    } catch {
+      case NonFatal(ex) =>
+        throw QueryCompilationErrors.cannotFindDescriptorFileError(filePath, ex)
```

Comment:
`readFileToByteArray` can throw a few different exceptions:
https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FileUtils.html#readFileToByteArray-java.io.File-

```
NullPointerException - if file is null.
FileNotFoundException - if the file does not exist, is a directory rather than a regular file, or for some other reason cannot be opened for reading.
IOException - if an I/O error occurs.
```
If it's IOException, it doesn't mean the descriptor file cannot be found, right? We can either throw a different error, or same error but with more generic error message/code

Reply:
The Javadoc there is incomplete. In github CI tests, this method throws `java.nio.file.NoSuchFileException`. Updated the PR.

```
sbt.ForkMain$ForkError: java.nio.file.NoSuchFileException: /non/existent/path.desc
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
	at java.nio.file.Files.newByteChannel(Files.java:361)
	at java.nio.file.Files.newByteChannel(Files.java:407)
	at java.nio.file.Files.readAllBytes(Files.java:3152)
	at org.apache.commons.io.FileUtils.readFileToByteArray(FileUtils.java:2581)
	at org.apache.spark.sql.protobuf.utils.ProtobufUtils$.readDescriptorFileContent(ProtobufUtils.scala:232)
```